profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LibrariesInfoCache.cpp

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "LibrariesInfoCache.h"
+
+#include "Log.h"
+
+#include <string.h>
+
+LibrariesInfoCache* LibrariesInfoCache::s_instance = nullptr;
+
+// This function is exposed by the ld_preload wrapper library.
+// This allows us to know if we have to reload the cache or not
+extern "C" unsigned long long dd_nb_calls_to_dlopen_dlclose() __attribute__((weak));
+
+LibrariesInfoCache::LibrariesInfoCache()
+{
+    LibrariesInfo.reserve(100);
+    NbCallsToDlopenDlclose = 0;
+    s_instance = this;
+    unw_set_iterate_phdr_function(unw_local_addr_space, LibrariesInfoCache::DlIteratePhdr);
+}
+
+LibrariesInfoCache* LibrariesInfoCache::Get()
+{
+    static LibrariesInfoCache Instance;
+    return &Instance;
```

Comment:
Why not `LibrariesInfoCache::s_instance`?

Reply:
in fact, I did not use this syntax since it's static from a 2-line function. And actually non found of `s_xxx` nor `g_xxx`

Reply:
got your comment. thx